tests/linux/test_platform_linux.cpp

```diff
@@ -70,6 +71,14 @@ struct PlatformLinux : public mpt::TestWithMockedBinPath
     template <typename VMFactoryType>
     void aux_test_driver_factory(const QString& driver)
     {
+        mpt::MockDNSMasqServerFactory::GuardedMock dnsmasq_server_factory_attr{
+            mpt::MockDNSMasqServerFactory::inject<NiceMock>()};
+        mpt::MockDNSMasqServerFactory* mock_dnsmasq_server_factory = dnsmasq_server_factory_attr.first;
+
+        ON_CALL(*mock_dnsmasq_server_factory, make_dnsmasq_server).WillByDefault([this](auto...) {
+            return std::make_unique<mpt::MockDNSMasqServer>();
+        });
+
```

Comment:
```
        mpt::MockDNSMasqServerFactory* mock_dnsmasq_server_factory = dnsmasq_server_factory_attr.first;

        ON_CALL(*mock_dnsmasq_server_factory, make_dnsmasq_server).WillByDefault([this](auto...) {
            return std::make_unique<mpt::MockDNSMasqServer>();
        });
```
I do not think this part is needed, basically `QemuVirtualMachineFactory` object has `qemu_platform`, and `qemu_platform_details` which is the instance of `qemu_platform` on linux has `DNSMasqServer` pointer. If we just mock the `DNSMasqServerFactory` and do nothing, then this [line](https://github.com/canonical/multipass/blob/f11f181fd7d8b46ec3b39bd94c9ffd3aa14ebc9a/src/platform/backends/qemu/linux/qemu_platform_detail_linux.cpp#L108) call returns the default value of the return type which is `nulltpr`. The code above makes it returns an pointer of `mpt::MockDNSMasqServer`, but the unit tests in this file did not use any of his mocked behavior. As a matter of fact, only the `QemuVirtualMachineFactory` constructor gets called, so it is ok to leave `DNSMasqServer` pointer empty.